build/BUILD.zlib

```diff
@@ -0,0 +1,194 @@
+# This config closely follows the original GN build file
+# Ref: https://chromium.googlesource.com/chromium/src/third_party/zlib.git/+/refs/heads/main/BUILD.gn
+# The main difference is that we don't set ZLIB_DLL because it messes things up on Windows
+
+package(default_visibility = ["//visibility:public"])
+
+zlib_warnings = [
+    "-Wno-incompatible-pointer-types",
+    "-Wunused-variable",
+]
+
+cc_library(
+    name = "zlib_common_headers",
+    hdrs = [
+        "chromeconf.h",
+        "deflate.h",
+        "inffast.h",
+        "inffixed.h",
+        "inflate.h",
+        "inftrees.h",
+        "zconf.h",
+        "zlib.h",
+        "zutil.h",
+    ],
+    defines = ["ZLIB_IMPLEMENTATION"],
+    includes = ["."],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "x86_defines",
+    defines = select({
+        "@platforms//os:windows": ["X86_WINDOWS"],
+        "//conditions:default": ["X86_NOT_WINDOWS"],
+    }),
+)
+
+cc_library(
+    name = "zlib_adler32_simd",
+    srcs = [
+        "adler32_simd.c",
+        "adler32_simd.h",
+    ],
+    copts = select({
+        "@platforms//cpu:x86_64": ["-mssse3"],
+        "//conditions:default": [],
+    }),
+    defines = select({
```

Comment:
These should really be `local_defines` – defines are added to every target that depends on zlib otherwise, which is really polluting the global defines namespace and unnecessarily making the command line of workerd longer. https://bazel.build/reference/be/c-cpp#cc_library.local_defines

Reply:
I agree; the main reason I left it all as define is to match BUILD.gn. I can take a crack at limiting the scope in another PR.

Reply:
There's also another nuisance: I'm using `defines` to avoid nested selects which don't work in bazel. I am actually not entirely sure how to get around this.

Reply:
Great question! We use `selects.config_setting_group` (https://bazel.build/docs/configurable-attributes#and-chaining) for this. Started with a PR on converting those.